### PR TITLE
feat: MCF endpoint

### PR DESCRIPTION
**Pull request by MiroCillik**
Pridany endpoint pre "Multi Channel Funnels" 
https://developers.google.com/analytics/devguides/reporting/mcf/v3/

V testoch zatial testujem len hlavicku vystupneho CSV, lebo na nasom webe (ktory sa pouziva v testoch) som to "MCF" nastavil len teraz.

A na update na PHP 7.2. coding style mam samostatnu issue https://github.com/keboola/google-analytics-extractor/issues/41 ;)

### CODE REVIEW

**Comment by MiroCillik** [comment]
Antisampling zatial v UI pre MCF nebude. Takze to ani netestujem.

File: src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php [hunk 1]
```diff
@@ -59,56 +60,85 @@ public function getCustomMetrics($accountId, $webPropertyId)
...
-     *   - dimensions - array of dimensions [OPTIONAL]
-     *   - filtersExpression - filter expression [OPTIONAL]
-     *   - segments - segment ID [OPTIONAL]
-     *   - dateRanges - array of Date ranges
-     *   - orderBy - dimension or metric to order by
-     *
-     * @return array
-     * @throws \Keboola\Google\ClientBundle\Exception\RestApiException
-     */
+    public function getReport($query)
```
Comment (odinuv) [inline src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php]: nemelo by se spis tohle jmenovat getBatch a to getBatch jmenovat getReport?

File: src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php [hunk 2]
```diff
@@ -195,6 +225,58 @@ private function processResponse($response, $query)
...
+            }
+
+            $dataSet[] = new Result($metrics, $dimensions);
+        }
+
+        $processed = [
+            'data' => $dataSet,
+            'query' => $query,
+            'totals' => $response['totalsForAllResults'],
+            'rowCount' => isset($response['totalResults'])?$response['totalResults']:0
```
Comment (odinuv) [inline src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php]: CS

File: src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php [hunk 3]
```diff
@@ -195,6 +225,58 @@ private function processResponse($response, $query)
         return $processed;
     }
 
+    private function processResponseMCF($response, $query)
+    {
+        if (empty($response['rows'])) {
+            return null;
```
Comment (odinuv) [inline src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php]: prijde me pitomy, ze to vraci null a pritom tady https://github.com/keboola/google-analytics-extractor/blob/master/src/Keboola/GoogleAnalyticsExtractor/Extractor/Antisampling.php#L49 se ocekava, ze je to pole.
Uz to tak bylo teda i predtim a jako funguje to, ale je to imho casovana bomba.

File: src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Result.php [hunk 4]
```diff
@@ -134,4 +134,10 @@ public function getDateFormatted()
 
         return $result;
     }
+
+    private function getDateKey($dimensions)
```
Comment (odinuv) [inline src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Result.php]: tak to taky muze byt static

File: src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php (hunk 1 above)
Comment (MiroCillik) [reply src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php]: Mozno mas pravdu, ja som to bral tak, ze ta funkcia getBatch() robi request na `*:batchGet` aj ked sa ta konstanta vola REPORTS_URL :)

https://github.com/keboola/google-analytics-extractor/pull/40/files/8d4c95e2d1b05c00e5648e76315396e8d2939c19#diff-9577aef28caef4c01677a5d2c81b9fddR17

Este sa nad tym zamyslim ;)

File: src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php (hunk 3 above)
Comment (MiroCillik) [reply src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php]: Jo, tiez som sa nad tym pozastavil, opravim to.

File: src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Result.php (hunk 4 above)
Comment (MiroCillik) [reply src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Result.php]: muze

**Comment by MiroCillik** [comment]
Diky Odin za review, upravil som to.

**Pull request merged**